src/gpuarray_buffer_cuda.c

```diff
@@ -937,41 +1021,123 @@ static void *call_compiler(const char *src, size_t len, const char *arch_arg,
                             2,
 #endif
                             opts);
-  if (log != NULL) {
-    err2 = nvrtcGetProgramLogSize(prog, &buflen);
-    if (err2 != NVRTC_SUCCESS) goto end2;
-    buf = malloc(buflen);
-    if (buf == NULL) goto end2;
-    err2 = nvrtcGetProgramLog(prog, (char *)buf);
-    if (err2 != NVRTC_SUCCESS) goto end2;
-    if (log_len != NULL) *log_len = buflen;
-    *log = (char *)buf;
-    buf = NULL;
+  if (nvrtcGetProgramLogSize(prog, &buflen) == NVRTC_SUCCESS) {
+    strb_appends(log, "NVRTC compile log::\n");
+    if (strb_ensure(log, buflen) == 0)
```

Comment:
We are cutting the final '\0' of the log at line 1028 so there will be enough space.

Also even if there wasn't enough space strb_appendc() will extend the buffer if required.